src/graphnet/data/dataconverter.py

```diff
@@ -427,11 +427,13 @@ def _extract_data(self, fileset: FileSet) -> List[OrderedDict]:
         while i3_file_io.more():
             try:
                 frame = i3_file_io.pop_physics()
+                assert self._is_not_null_split_frame(frame)
```

Comment:
As I mention [here](https://github.com/graphnet-team/graphnet/pull/432#issuecomment-1433574619), I am not particularly fond of this solution, and would propose (1) amore general frame-skipping method and (2) separating this logic from whether the frame is a P-frame. But the proposed code works and is not unreasonable, so I'll leave it at your discretion, @Aske-Rosted.

Reply:
@Aske-Rosted I think Andreas suggestion is better than what I mentioned earlier.

Reply:
Is the latest commit satisfactory? I am not 100% certain that all exceptions coming from the I3 modules will contain I3 in the exception name, but I would hope that they do...